src/completion.rs

```diff
@@ -0,0 +1,38 @@
+use crate::cli::CliArgs;
+use async_trait::async_trait;
+use clap::{CommandFactory, Parser};
+use clap_complete::{generate, Shell};
+// use color_eyre::Result;
+use anyhow::Result;
+use std::io::stdout;
+use std::process::ExitCode;
+
+/// Prints completion for shells to use.
+#[derive(Parser)]
+pub(crate) struct CompletionSubcommand {
+    /// Shell
+    #[arg(value_enum)]
+    shell: Shell,
+}
```

Comment:
Out of curiosity, is this module (and the struct/traits on it) based on existing template/code? Or something you wrote?

Reply:
In general, let's try to reduce boiler plate. Maybe this will help:

https://lib.rs/crates/clap_complete_command

Reply:
> In general, let's try to reduce boilerplate. Maybe this will help:
> 
> https://lib.rs/crates/clap_complete_command

Didn't see this. But anyways implemented something similar only.